lexicons/app/bsky/feed/threadgate.json

```diff
@@ -0,0 +1,43 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.feed.threadgate",
+  "defs": {
+    "main": {
+      "type": "record",
+      "key": "tid",
+      "description": "Defines interaction gating rules for a thread. The rkey of the threadgate record should match the rkey of the thread's root post.",
+      "record": {
+        "type": "object",
+        "required": ["post", "createdAt"],
+        "properties": {
+          "post": { "type": "string", "format": "at-uri" },
+          "allow": {
+            "type": "array",
+            "maxLength": 5,
+            "items": {
+              "type": "union",
+              "refs": ["#mentionRule", "#followingRule", "#listRule"]
+            }
+          },
+          "createdAt": { "type": "string", "format": "datetime" }
+        }
+      }
+    },
+    "mentionRule": {
+      "type": "object",
+      "description": "Allow replies from actors mentioned in your post."
+    },
+    "followingRule": {
+      "type": "object",
+      "description": "Allow replies from actors you follow."
+    },
```

Comment:
Aren't `object` types supposed to have 1 or more `properties`? I maintain a lexicon codegen plugin and I noticed it's choking on these new types

Reply:
Ah looks like it was already reported: https://github.com/bluesky-social/atproto/issues/1615